### PR TITLE
Merge SANBI changes to TB Profiler

### PR DESCRIPTION
**Pull request by pvanheus**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by pvanheus** [comment]
This PR adds:
* fastq.gz support
* Paired collection input support
* BAM input support
* 1 extra test
* PDF output support

**Comment by Kurt-Hanel** [comment]
@pvanheus Does this add support for lists of pairs as well?

**Comment by pvanheus** [comment]
Yes, Galaxy will support that automatically.

File: tools/tb-profiler/tb_profiler_profile.xml [hunk 1]
```diff
@@ -4,82 +4,164 @@
...
+                    <option value="high">High</option>
+                    <option value="optimise">Optimise</option>
+                </param>
+                <param label="Minimum coverage fraction to infer deletion" type="float"
+                        help="Used to infer a deletion if the fraction of a gene covered falls below this value."
+                        argument="--min_gene_frac" value="0.9" />
+
+                <param name="min_depth" label="Min Depth" type="integer" value="10" help="Minimum depth required to call variant. Bases with depth below this cutoff will be marked as missing (default: 10)"/>
+                <param name="mapper" label="Mapper" type="select" help="Mapping tools to use (default: bwa)">
+                    <option value="bwa" selected="true">bwa</option>
```
Comment (abretaud) [inline tools/tb-profiler/tb_profiler_profile.xml]: Should these mappers be in the requirements?

File: tools/tb-profiler/tb_profiler_profile.xml [hunk 2]
```diff
@@ -4,82 +4,164 @@
...
+        #if $output_format == "pdf"
+            --pdf
+        #else if $output_format == "txt"
+            --txt
+        #end if
+        && mv results/tbprofiler.results.json $results_json
+        #if str($fastq_or_bam.input_select) != "bam"
+            && mv bam/tbprofiler.bam $output_bam
+        #end if
+        && bcftools view -Ov -o$output_vcf vcf/tbprofiler.targets.csq.bcf
```
Comment (abretaud) [inline tools/tb-profiler/tb_profiler_profile.xml]: quotes around $output_vcf

File: tools/tb-profiler/tb_profiler_profile.xml (hunk 1 above)
Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_profile.xml]: They are requirements of the tbprofiler conda package and thus they come in when that is installed.

**Comment by bgruening** [comment]
Thanks!

**Pull request merged**